Objects/exceptions.c

```diff
@@ -944,10 +978,15 @@ exceptiongroup_split_check_match(PyObject *exc,
         return is_true;
     }
     case EXCEPTION_GROUP_MATCH_INSTANCES: {
-        if (PySequence_Check(matcher_value)) {
-            return PySequence_Contains(matcher_value, exc);
+        assert(PySet_Check(matcher_value));
+        if (! _PyBaseExceptionGroup_Check(exc)) {
+            int include = PySet_Contains(matcher_value, exc);
+            if (include == -1) {
+                return -1;
+            }
+            return include;
```

Comment:
Are we planning on adding more error handling here? If not, we can just pass the return value of `PySet_Contains` straight to the caller.

```suggestion
            return PySet_Contains(matcher_value, exc);
```